code/modules/power/engines/tesla/tesla_zap.dm

```diff
@@ -0,0 +1,166 @@
+/*
+Alright so, this is a file that has the proc for a proc that works like 90% of all zaps.
+basic_tesla_zap() can be called by anything and does not need any overrides.
+*/
+
+//Zap constants, speeds up targeting
+#define COIL (ROD + 1)
```

Comment:
if you are going to keep both copies of tesla_zap around (which as below i'm not sure is necessary) i'd pull these constants up into __DEFINES